superset/db_engine_specs/postgres.py

```diff
@@ -296,3 +297,17 @@ def get_column_spec(  # type: ignore
         return super().get_column_spec(
             native_type, column_type_mappings=column_type_mappings
         )
+
+    @classmethod
+    def get_cancel_query_payload(cls, cursor: Any, query: Query) -> Any:
+        cursor.execute("SELECT current_user")
+        row = cursor.fetchone()
+        return row[0]
+
+    @classmethod
+    def cancel_query(cls, cursor: Any, query: Query, payload: Any) -> None:
+        cursor.execute(
+            "SELECT pg_terminate_backend(pid) "
+            "FROM pg_stat_activity "
+            "WHERE pid <> pg_backend_pid() and usename='%s'" % payload
```

Comment:
```suggestion
            "WHERE pid <> pg_backend_pid() and username='%s'" % payload
```

Won't this kill all user queries in progress?

Reply:
thanks, fixed by 0f4947160ff0f51b76aa3b0f4d8fb79d6cbcd008